src/sys/prctl.rs

```diff
@@ -0,0 +1,229 @@
+    }
+    impl TryFrom<i32>
+}
+
+fn prctl_set_bool(option: PrctlOption, status: bool) -> Result<()> {
+    let res = unsafe { libc::prctl(option as c_int, status as c_ulong, 0, 0, 0) };
+    Errno::result(res).map(drop)
+}
+
+/// Set the "child subreaper" attribute for this process
+pub fn set_child_subreaper(attribute: bool) -> Result<()> {
+    prctl_set_bool(PR_SET_CHILD_SUBREAPER, attribute)
+}
+
+/// Get the "child subreaper" attribute for this process
+pub fn get_child_subreaper() -> Result<bool> {
+    // prctl writes into this var
+    let mut subreaper: c_int = 0;
+    let subreaper_ptr: *mut c_int = &mut subreaper;
+
+    let res = unsafe {
+        libc::prctl(PR_GET_CHILD_SUBREAPER as c_int, subreaper_ptr as c_ulong, 0, 0, 0)
+    };
+
+    Errno::result(res).map(|_| subreaper != 0)
+}
+
+/// Set the dumpable attribute which determines if core dumps are created for this process.
+pub fn set_dumpable(attribute: bool) -> Result<()> {
+    prctl_set_bool(PR_SET_DUMPABLE, attribute)
+}
+
+/// Get the dumpable attribute for this process.
+pub fn get_dumpable() -> Result<bool> {
+    let res = unsafe {
+        libc::prctl(PR_GET_DUMPABLE as c_int, 0, 0, 0, 0)
+    };
+
+    Errno::result(res).map(|res| res != 0)
+}
+
+/// Set the "keep capabilities" attribute for this process. This causes the thread to retain
+/// capabilities even if it switches its UID to a nonzero value.
+pub fn set_keepcaps(attribute: bool) -> Result<()> {
+    prctl_set_bool(PR_SET_KEEPCAPS, attribute)
+}
+
+/// Get the "keep capabilities" attribute for this process
+pub fn get_keepcaps() -> Result<bool> {
+    let res = unsafe {
+        libc::prctl(PR_GET_KEEPCAPS as c_int, 0, 0, 0, 0)
+    };
+
+    Errno::result(res).map(|res| res != 0)
+}
+
+/// Clear the thread memory corruption kill policy and use the system-wide default
+pub fn clear_mce_kill() -> Result<()> {
+    let res = unsafe {
+        libc::prctl(PR_MCE_KILL as c_int, libc::PR_MCE_KILL_CLEAR, 0, 0, 0)
+    };
+
+    Errno::result(res).map(drop)
+}
+
+/// Set the thread memory corruption kill policy
+pub fn set_mce_kill(policy: PrctlMCEKillPolicy) -> Result<()> {
+    let res = unsafe {
+        libc::prctl(PR_MCE_KILL as c_int, libc::PR_MCE_KILL_SET, policy as c_ulong, 0, 0)
+    };
+
+    Errno::result(res).map(drop)
+}
+
+/// Get the thread memory corruption kill policy
+pub fn get_mce_kill() -> Result<PrctlMCEKillPolicy> {
+    let res = unsafe {
+        libc::prctl(PR_MCE_KILL_GET as c_int, 0, 0, 0, 0)
+    };
+
+    Errno::result(res).map(|res| PrctlMCEKillPolicy::try_from(res).unwrap())
+}
+
+/// Manipulates various aspects of the behaviour of the calling process or thread.
+///
+/// # Safety
+/// Note that careless use of some prctl() operations can confuse the user-space run-time
+/// environment, so these operations should be used with care.
+///
+/// For more information, see [prctl(2)](https://man7.org/linux/man-pages/man2/prctl.2.html).
+pub unsafe fn prctl(
```

Comment:
I don't think there's any reason to leave this in here.  It's no better than calling libc directly.

Reply:
Agreed, I have removed it now.